File: app/src/main/java/com/bitvale/telegramcharts/ChartPagerAdapter.java
package com.bitvale.telegramcharts;

import androidx.annotation.NonNull;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentPagerAdapter;
import com.bitvale.chartview.model.Chart;

import java.util.ArrayList;

/**
 * Created by Alexander Kolpakov (jquickapp@gmail.com) on 17-Mar-19
 */
public class ChartPagerAdapter extends FragmentPagerAdapter {

    private ArrayList<Chart> chartList;

    ChartPagerAdapter(@NonNull FragmentManager fm, ArrayList<Chart> chartList) {
        super(fm);
        this.chartList = chartList;
    }

    @NonNull
    @Override
    public Fragment getItem(int position) {
        return ChartFragment.newInstance("Chart " + (position + 1), chartList.get(position));
    }

    @Override
    public int getCount() {
        return chartList.size();
    }
}

File: app/src/main/java/com/bitvale/telegramcharts/ChartPagerAdapter.kt
package com.bitvale.telegramcharts

import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentPagerAdapter
import com.bitvale.chartview.model.Chart

/**
 * Created by Alexander Kolpakov (jquickapp@gmail.com) on 11-Mar-19
 */
class ChartPagerAdapter(fragmentManager: FragmentManager, private val chartList: ArrayList<Chart>)
    : FragmentPagerAdapter(fragmentManager) {

    override fun getCount(): Int = chartList.size

    override fun getItem(position: Int): Fragment {
        return ChartFragment.newInstance("Chart ${position + 1}", chartList[position])
    }
}